[RFC] [commands] custom default arguments (#1849)
Rapptz/discord.py: docs/ext/commands/commands.rst

The first parameter of the error handler is the :class:`.Context` while the second one is an exception that is derived from
:exc:`~ext.commands.CommandError`. A list of errors is found in the :ref:`ext_commands_api_errors` page of the documentation.

Custom Defaults
---------------

Custom defaults allow us to specify :class:`.Context`-based defaults. Custom defaults are always classes which inherit from
:class:`.CustomDefault`.

> The library provides some simple default implementations in the :module:`default` - :class:`default.Author`, :class:`default.Channel`,

Reviewer: ```suggestion
The library provides some simple default implementations in ``ext.commands.default`` - :class:`.default.Author`, :class:`.default.Channel`,
```

`:module:` is not a valid tag... Is this necessary? (won't make a difference if `:mod:` is used).

> and :class:`default.Guild` returning the corresponding properties from the Context. These can be used along with Converters to

Reviewer: ```suggestion
and :class:`.default.Guild` returning the corresponding properties from the Context. These can be used along with Converters to
```

> simplify your individual commands. You can also use :class:`default.Call` to quickly wrap existing functions.

Reviewer: ```suggestion
simplify your individual commands. You can also use :class:`.default.Call` to quickly wrap existing functions.
```

> A DefaultParam returning `None` is valid - if this should be an error, raise :class:`.MissingRequiredArgument`.

Reviewer: ```suggestion
A DefaultParam returning ``None`` is valid - if this should be an error, raise :class:`.MissingRequiredArgument`.
```

> .. code-block:: python3

Reviewer: I feel like specific lines should be highlighted for this as it is a long codeblock.

Reviewer 2: Btw how do you highlight them

Reviewer: ```rst
.. code-block:: python3
    :emphasize-lines: 1,2
        
        print('hello world!')
        print('test.')
```
